nomad/structs/structs.go

```diff
@@ -1269,15 +1284,17 @@ func (r *Resources) Canonicalize() {
 
 // MeetsMinResources returns an error if the resources specified are less than
 // the minimum allowed.
+// This is based on the minimums defined in the Resources type
 func (r *Resources) MeetsMinResources() error {
 	var mErr multierror.Error
-	if r.CPU < 20 {
+	minResources := MinResources()
+	if r.CPU < minResources.CPU {
 		mErr.Errors = append(mErr.Errors, fmt.Errorf("minimum CPU value is 20; got %d", r.CPU))
```

Comment:
This hard codes 20 but should use the value from the min resources. Same for all others.
`fmt.Errorf("minimum CPU value is %d; got %d", minResources.CPU, r.CPU)`